### PR TITLE
#3258 Deps Upgrades #2

### LINKED ISSUE #3258: Dependencies upgrade
## The suggestion

Need to upgrade dependencies. I suspect it might be a good idea to separate this into multiple issues, if there are real code changes required when upgrading react-native.

**Why?**
Cause we're very behind.

## Describe alternatives you've considered
N/A

## Additional context
N/A

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #3258

## Change summary

- Adds some of the smaller native module upgrades

## Testing

- [ ] App still worky

### Related areas to think about

N/A

### CODE REVIEW

File: src/actions/FinaliseActions.js [hunk 1]
```diff
@@ -36,7 +34,7 @@ const finalise = () => (dispatch, getState) => {
   try {
     UIDatabase.write(() => finaliseItem.finalise(UIDatabase, currentUser));
   } catch (error) {
-    bugsnagClient.notify(error);
+    Bugsnag.notify(error);
```
Comment (joshxg) [inline src/actions/FinaliseActions.js:37]: Just a singleton now don't need to make the client

File: src/main.js [hunk 2]
```diff
@@ -15,6 +16,8 @@ import { backHandler, rootNavigatorRef } from './navigation';
 
 import MSupplyMobileApp from './mSupplyMobileApp';
 
+Bugsnag.start();
```
Comment (joshxg) [inline src/main.js:19]: Just need to start the singleton in root

File: src/sync/Synchroniser.js [hunk 3]
```diff
@@ -60,6 +58,8 @@ export class Synchroniser {
     this.extraHeaders = {
       'msupply-site-uuid': DeviceInfo.getUniqueId(),
       'mobile-version': mobileVersion,
+      'Content-Type': 'application/json',
+      Accept: 'application/json',
```
Comment (joshxg) [inline src/sync/Synchroniser.js:62]: Needed to add this actually because of the node bump: I suppose `fetch` impl changed and needed these to send `POST`

**Review by wlthomson — APPROVED** [review]
LGTM.

**Review by mark-prins — APPROVED** [review]
LGTM too!

**Pull request merged**